src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs

```diff
@@ -137,6 +137,11 @@ public static AttributeCollection FromExisting(AttributeCollection existing, par
         {
             get
             {
+                if (attributeType == null)
```

Comment:
This is a virtual so once we nullable annotate this, all the overrides won't be able to accept null. It seems like they endup throwing because they try to access a dictionary with a null key, but just want to make sure we consider this. 

Also, we would need to update the docs: https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.attributecollection.item?view=netcore-3.1

Reply:
Yes this is deliberate. Not sure if you saw the comment but basically if `attributeType` was null, then we'd actually get a non-zero index, as the code is:
```cs
// 2 passes here for perf. Really!  first pass, we just
// check equality, and if we don't find it, then we
// do the IsAssignableFrom dance. Turns out that's
// a relatively expensive call and we try to avoid it
// since we rarely encounter derived attribute types
// and this list is usually short.
if (_foundAttributeTypes == null)
{
    _foundAttributeTypes = new AttributeEntry[FoundTypesLimit];
}

int ind = 0;

for (; ind < FoundTypesLimit; ind++)
{
    if (_foundAttributeTypes[ind].type == attributeType)
    {
        int index = _foundAttributeTypes[ind].index;
        if (index != -1)
        {
            return Attributes[index];
        }
        else
        {
            return GetDefaultAttribute(attributeType);
        }
    }
    if (_foundAttributeTypes[ind].type == null)
        break;
}
```

`_foundAttributeTypes[ind].type == attributeType)` would be true because everything is initialized internally to `null`. So therefore, when we do `return Attributes[index];` this throws `IndexOutOfRangeException` because the `_foundAttributeTypes` contains more elements (initialized to null) than `Attributes.Count`

Reply:
See https://github.com/dotnet/runtime/pull/36118#discussion_r422224661

We could consider just using `List<Attribute>` but I didn't want to make these sorts of changes here